app/state_machines/form_state_machine.rb

```diff
@@ -0,0 +1,49 @@
+module FormStateMachine
+  extend ActiveSupport::Concern
+
+  included do
+    include AASM
+
+    enum :state, {
+      draft: "draft",
+      deleted: "deleted",
+      live: "live",
+      draft_live: "draft_live",
+    }
+
+    aasm column: :state, enum: true do
+      state :draft, initial: true
+      state :deleted, :live, :draft_live
+
+      event :delete_form do
+        after do
+          destroy!
+        end
+
+        transitions from: :draft, to: :deleted
+      end
+
+      event :make_live do
+        after do
+          live_at ||= Time.zone.now
+          touch(time: live_at)
+
+          form_blob = snapshot(live_at:)
+          made_live_forms.create!(json_form_blob: form_blob.to_json, created_at: live_at)
+        end
+
+        transitions from: %i[draft draft_live], to: :live, guard: proc { task_status_service.mandatory_tasks_completed? }
+      end
+
+      event :draft_new_live_form do
```

Comment:
Can I suggest we call this `create_draft` to match the language used in the 'Create a draft to edit' button?

Reply:
Its open for change and I dont have strong opinions on the event names.

🤔 , could this be confusing to think a new form is  being created because "draft" is the initial state?

Reply:
related https://github.com/alphagov/forms-api/pull/414#discussion_r1465099113

Reply:
I'm in favour of this suggestion on the same reasoning as the comment on that other PR - it makes sense to me for the transition name to match how the user sees the transition. Not set on it though Al so happy to leave it up to you!

I do agree that there may be some confusion, but I think that confusion is probably unavoidable because the draft/live logic in our code doesn't match the way we present that logic to users. In the longer term I thinik we probably need to resolve that difference, but I appreciate that's much bigger than this PR

Reply:
@lfdebrux @DavidBiddle  what about `create_draft_from_live_form`? We now have `live_with_draft` as a status

Reply:
other potentials options
- `make_changes_to_live_form`
- `make_draft_from_live_form`
- `add_draft` (not sure about this one makes it sounds like we are creating another new form record)
-

Reply:
I like create_draft_from_live_form/make_draft_from_live_form

Reply:
`create_draft_from_live_form` works for me.